cmd/ctl/pkg/upgrade/migrate/migrate.go

```diff
@@ -0,0 +1,280 @@
+	acmeinstall.Install(scheme)
+}
+
+// Options is a struct to support renew command
+type Options struct {
+	genericclioptions.IOStreams
+	*factory.Factory
+
+	client client.Client
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		IOStreams: ioStreams,
+	}
+}
+
+// NewCmdMigrate returns a cobra command for updating resources in an apiserver
+// to force a new storage version to be used.
+func NewCmdMigrate(ctx context.Context, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	o := NewOptions(ioStreams)
+	cmd := &cobra.Command{
+		Use:     "migrate",
+		Short:   "Migrate all existing persisted cert-manager resources to the v1 API version",
+		Long:    long,
+		Example: example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Validate(cmd, args))
+			cmdutil.CheckErr(o.Complete())
+			cmdutil.CheckErr(o.Run(ctx, args))
+		},
+	}
+
+	o.Factory = factory.New(ctx, cmd)
+
+	return cmd
+}
+
+// Validate validates the provided options
+func (o *Options) Validate(cmd *cobra.Command, args []string) error {
+	return nil
+}
+
+// Complete takes the command arguments and factory and infers any remaining options.
+func (o *Options) Complete() error {
+	var err error
+	o.client, err = client.New(o.RESTConfig, client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run executes renew command
+func (o *Options) Run(ctx context.Context, args []string) error {
+	// Check all cert-manager CRDs to ensure that they have their
+	// storage version set to v1.
+	allCRDNames := []string{
+		"certificates.cert-manager.io",
+		"certificaterequests.cert-manager.io",
+		"issuers.cert-manager.io",
+		"clusterissuers.cert-manager.io",
+		"orders.acme.cert-manager.io",
+		"challenges.acme.cert-manager.io",
+	}
+
+	fmt.Fprintln(o.Out, "Checking all cert-manager CustomResourceDefinitions have storage version set to 'v1'")
+	allV1, crdsRequiringMigration, err := o.ensureCRDStorageVersionEquals(ctx, "v1", allCRDNames)
+	if err != nil {
+		return err
+	}
+	if !allV1 {
+		fmt.Fprintln(o.ErrOut, "It looks like you are running a pre-1.0 version of cert-manager. Please upgrade cert-manager to v1.6 before upgrading to v1.7.")
+		return fmt.Errorf("migration failed")
+	}
+	fmt.Fprintln(o.Out, "All CustomResourceDefinitions have 'v1' configured as the storage version.")
+
+	/*
+		fmt.Fprintln(o.Out, "Looking for CRDs that contain resources that require migrating to 'v1'...")
+		crdsRequiringMigration, err := o.discoverCRDsRequiringMigration(ctx, "v1", allCRDNames)
+		if err != nil {
+			fmt.Fprintf(o.ErrOut, "Failed to determine resource types that require migration: %v\n", err)
+			return err
+		}
+		if len(crdsRequiringMigration) == 0 {
+			fmt.Fprintln(o.Out, "Nothing to do. cert-manager CRDs do not have 'status.storedVersions' containing old API versions. You may proceed to upgrade to cert-manager v1.7.")
+			return nil
+		}
+	*/
```

Comment:
This is the code that I think we want to remove, which would cause the tool to only run if `status.storedVersions` is not set to `v1`.